packages/jest-haste-map/src/__tests__/index.test.js

```diff
@@ -346,6 +346,14 @@ describe('HasteMap', () => {
     expect(hasteFS.matchFiles('.git')).toEqual([]);
   });
 
+  it('ignores sapling vcs directories without ignore pattern', async () => {
+    mockFs[path.join('/', 'project', 'fruits', '.sl', 'package.json')] = `
+      invalid}{
+    `;
+    const {hasteFS} = await (await HasteMap.create(defaultConfig)).build();
+    expect(hasteFS.matchFiles('.sl')).toEqual([]);
```

Comment:
without the patch in `index.ts` this throws an error about "could not parse invalid JSON"